curriculum/challenges/english/08-coding-interview-prep/rosetta-code/run-length-encoding.md

```diff
@@ -0,0 +1,128 @@
+---
+id: 5eb257ba998e23057d209754
+title: Run-length Encoding
+challengeType: 5
+---
+
+## Description
+<section id='description'>
+Given a string containing uppercase characters (A-Z), compress repeated 'runs' of the same character by storing the length of that run, and provide a function to reverse the compression.
+</section>
+
+## Instructions
+<section id='instructions'>
+Write two functions <code>encode</code> and <code>decode</code>.
+
+<code>encode(input)</code> takes a string and return the run-length encoding as an array
+containing pairs with the first element representing the number of occurences and the second
+element representing the letter.
+
+<code>decode(input)</code> takes an array of pairs representing a run-length encoded string
+and returns the decoded string.
+
+You may assume that the input will only contain uppercase letters.
+
+Example:
+<br>
+Original: <code>ABBBCDDDD</code>
+<br>
+Encoded: <code>[[1,'A'], [3,'B'], [1,'C'], [4,'D']]</code>
+
+</section>
```

Comment:
Tried to play with the wording a bit here to make things just a little be clearer. Another thing we can do is adjust the spacing a bit here to take advantage of markdown-style code highlighting with backticks instead of `<code>` tags:

````
<section id='instructions'>

Write two functions, `encode` and `decode`.

`encode(input)` takes a string and returns the run-length encoding as a 2D array. Each element in the returned array is an array containing pairs, with the first element representing the number of occurrences and the second element representing the letter.

`decode(input)` takes a 2D array of pairs representing a run-length encoded string and returns the decoded string.

You may assume that the input will only contain uppercase letters.
</section>
````